src/App.js

```diff
@@ -87,37 +88,101 @@ export class App extends Component {
           toggle={this.state.showMenu}
         />
         <Switch>
-          <Route exact path="/" component={FrontPage} onEnter={this.gotoTop} />
-          <Route path="/about" component={AboutPage} onEnter={this.gotoTop} />
-          <Route path="/blog" component={BlogPage} onEnter={this.gotoTop} />
-          <Route path="/blog/tags/:tags?" component={BlogPage} />
-          <Route path="/blog/:post" component={BlogPage} />
+          <Route exact path="/" 
+            render={(props) => {
+              this.gotoTop()
+              console.log("testing")
+              return <FrontPage {...props}/>
+            }} 
+          />
+          <Route 
+            path="/about" 
+            render={(props) => {
+              this.gotoTop()
+              console.log("testing")
+              return <AboutPage {...props}/>
+            }} 
```

Comment:
What about making a function that has component (here AboutPage) as parameter, then use `render={(props) => this.switchPage(AboutPage, props)}`